Build.c

```diff
@@ -188,10 +188,12 @@ int Build(struct mddev_ident *ident, struct mddev_dev *devlist, struct shape *s,
 		pr_err("array %s built and started.\n", chosen_name);
 	wait_for(chosen_name, mdfd);
 	close(mdfd);
+	close(bitmap_fd);
```

Comment:
```suggestion
if (s->bitmap_file) {
		int bitmap_fd = open(s->bitmap_file, O_RDWR);
		if (bitmap_fd < 0) {
			int major = BITMAP_MAJOR_HI;
			bitmapsize = s->size >> 9; /* FIXME wrong for RAID10 */
			if (CreateBitmap(s->bitmap_file, 1, NULL,
					 s->bitmap_chunk, c->delay,
					 s->write_behind, bitmapsize, major)) {
				goto abort;
			}
			bitmap_fd = open(s->bitmap_file, O_RDWR);
			if (bitmap_fd < 0) {
				pr_err("%s cannot be opened.\n", s->bitmap_file);
				goto abort;
			}
		}
		if (ioctl(mdfd, SET_BITMAP_FILE, bitmap_fd) < 0) {
			pr_err("Cannot set bitmap file for %s: %s\n", chosen_name,
				 strerror(errno));
			close(bitmap_fd);
			goto abort;
		}
		close(bitmap_fd);
	}
```

Reply:
Hi, 
please look at suggested change above , this is what I mean in my previous comments
"bitmap_fd, it is only needed inside in one "if" block" and "this check "if (bitmap_fd >= 0) {" is redundant , previous "if" block in case of bitmap_fd < 0 will go to abort:"

Reply:
added your code the patch.

Reply:
Strange... as it is in my version...  when I push a new version, it should be there.